src/core/Akka.Remote/Endpoint.cs

```diff
@@ -109,20 +109,17 @@ public DefaultMessageDispatcher(ActorSystem system, IRemoteActorRefProvider prov
                     var msgLog = $"RemoteMessage: {payload} to {recipient}<+{originalReceiver} from {sender}";
                     _log.Debug("received local message [{0}]", msgLog);
                 }
-                if (payload is ActorSelectionMessage)
+                if (payload is ActorSelectionMessage sel)
                 {
-                    var sel = (ActorSelectionMessage)payload;
-
-                    var actorPath = "/" + string.Join("/", sel.Elements.Select(x => x.ToString()));
```

Comment:
A minor performance improvement: we were allocating a new `actorPath` string every single time we received an `ActorSelectionMessage` in Akka.Remote even though its value is only used when `_settings.UntrustedMode` returns `true`, which it doesn't by default.

So instead, we no longer allocate this object unless that setting is enabled - we do have to allocate the string twice now, however, due to the way I moved that code into its own function. Not a huge deal given that untrusted mode isn't used all that much, but I wanted to call out this change on the PR.